cmd/git/main.go

```diff
@@ -423,9 +444,28 @@ func checkCredentials() (credentialType, error) {
 		return typeUsernamePassword, nil
 
 	case hasUsername && !hasPassword || !hasUsername && hasPassword:
-		return typeUndef, &ExitError{Code: 110, Message: "Basic Auth incomplete: Both username and password need to be configured"}
-
+		return typeUndef, &ExitError{
+			Code:    110,
+			Message: "Basic Auth incomplete: Both username and password need to be configured",
+			Reason:  shpgit.AuthInvalidUserOrPass,
+		}
 	}
 
 	return typeUndef, &ExitError{Code: 110, Message: "Unsupported type of credentials provided, either SSH private key or username/password is supported"}
 }
+
+func writeErrorResults(failure *shpgit.ErrorResult) (err error) {
+	if flagValues.resultErrorReason == "" || flagValues.resultErrorMessage == "" {
+		return nil
+	}
+
+	if err = ioutil.WriteFile(flagValues.resultErrorMessage, []byte(strings.TrimSpace(failure.Message)), 0600); err != nil {
```

Comment:
I see your point. In my mind there was no need for further writes, since step should only write the error results if the container terminates with non-zero. However, we can not assume that no step writes an error results in a perfectly fine container. Thus, I will change the rights to be more open.